src/mango_httpd.erl

```diff
@@ -50,8 +50,24 @@ handle_req_int(_, _) ->
 
 
 handle_index_req(#httpd{method='GET', path_parts=[_, _]}=Req, Db) ->
+    Params = lists:flatmap(fun({K, V}) -> parse_index_param(K, V) end,
+        chttpd:qs(Req)),
     Idxs = lists:sort(mango_idx:list(Db)),
-    JsonIdxs = lists:map(fun mango_idx:to_json/1, Idxs),
+    JsonIdxs0 = lists:map(fun mango_idx:to_json/1, Idxs),
+    % Set limit default to 200 here since dashboard defaults to 200 per page.
+    Limit = case couch_util:get_value(limit, Params, 200) of
+        Limit0 when Limit0 < 1 ->
+            ?MANGO_ERROR(invalid_list_index_params);
+        Limit0 ->
+            Limit0
+    end,
+    Skip = case couch_util:get_value(skip, Params, 0) of
+        Skip0 when Skip0 < 0; Skip0 > length(JsonIdxs0) ->
+            ?MANGO_ERROR(invalid_list_index_params);
+        Skip0 ->
+            Skip0
+    end,
+    JsonIdxs = lists:sublist(JsonIdxs0, Skip+1, Limit),
 	chttpd:send_json(Req, {[{indexes, JsonIdxs}]});
```

Comment:
Problem here: because there is no way to get total amount of indexes, client/Fauxton couldn't be sure how much pages it have to show in rule bar without requesting all of them and count them manually what makes further limit/skip requests useless since all the data is already loaded.